src/main/java/projectbookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,136 @@
+package projectbookstore.service.impl;
+
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import projectbookstore.dto.order.CreateOrderRequestDto;
+import projectbookstore.dto.order.OrderDto;
+import projectbookstore.dto.order.UpdateOrderStatusDto;
+import projectbookstore.dto.orderitem.OrderItemDto;
+import projectbookstore.exception.EntityNotFoundException;
+import projectbookstore.mapper.OrderItemMapper;
+import projectbookstore.mapper.OrderMapper;
+import projectbookstore.model.CartItem;
+import projectbookstore.model.Order;
+import projectbookstore.model.OrderItem;
+import projectbookstore.model.ShoppingCart;
+import projectbookstore.model.User;
+import projectbookstore.repository.OrderItemRepository;
+import projectbookstore.repository.OrderRepository;
+import projectbookstore.repository.ShoppingCartRepository;
+import projectbookstore.service.OrderService;
+import projectbookstore.service.ShoppingCartService;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartService shoppingCartService;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Transactional
+    @Override
+    public OrderDto save(CreateOrderRequestDto requestDto, User user) {
+        ShoppingCart shoppingCart = shoppingCartService.getExistedCart(user.getId());
+        Order order = createNewOrder(requestDto, user, shoppingCart);
+        Order savedOrder = orderRepository.save(order);
+        Set<OrderItem> orderItems = createOrderItems(savedOrder, shoppingCart);
+        savedOrder.setOrderItems(orderItems);
+        shoppingCartRepository.deleteById(shoppingCart.getId());
+        return orderMapper.toDto(savedOrder);
+    }
+
+    @Transactional
```

Comment:
get operations can be like that to improve performance a bit
```suggestion
    @Transactional(readOnly = true)
```